src/app/locator/qgsinbuiltlocatorfilters.cpp

```diff
@@ -362,8 +362,11 @@ void QgsAllLayersFeaturesLocatorFilter::prepare( const QString &string, const Qg
     req.setSubsetOfAttributes( expression.referencedAttributeIndexes( layer->fields() ).toList() );
     if ( !expression.needsGeometry() )
       req.setFlags( QgsFeatureRequest::NoGeometry );
+    QString enhancedSearch = string;
+    enhancedSearch.replace( " ", "%" );
```

Comment:
```suggestion
     enhancedSearch.replace( ' ', '%' );
```

The char based version should be more efficient